js-classes/inheritance-es5.js

```diff
@@ -0,0 +1,71 @@
+function Car(name, model, year, color, maxSpeed, fuelCapacity, fuelConsumption){
+	this.name = name;
+	this.model = model;
+	this.year = year;
+	this.color = color;
+	this.maxSpeed = maxSpeed;
+	this.fuelCapacity = fuelCapacity || 60;
+	this.fuelConsumption = fuelConsumption || 10;
+}
+
+Car.prototype.getFullName = function(name, model){
+	console.log(this.model + ' ' + this.name);
+}
+
+Car.prototype.getAge = function(){
+var date = new Date;
+var year = date.getFullYear();
+return year - this.year
+}
+
+Car.prototype.changeColor = function(color){
+	if (this.color == 'red') {
+		console.log('The color of your car is already red');
+	} else {
+		var newColor = 'green';
+		this.color = newColor;
+		console.log('Now your car is ' + newColor + ' color');
+	}
+}
+
+Car.prototype.calculateWay = function(kilometers, fuel){
+	if (fuel < 10){
+		console.log('You need to refuel');
+	};
+		var speed = this.maxSpeed;
+		var time = kilometers / speed;
+		console.log ('Your path will takes ' + time.toFixed(1) + ' hour');
+
+//	var fuelCapacityOnAllWay = (kilometers * this.fuelConsumption) / 100;
+//	var qualityCapacity = fuelCapacityOnAllWay / this.fuelCapacity;
+//	if 	(qualityCapacity > 1) {
+//			console.log('You need to refuel ' + qualityCapacity + ' times');	
+//	};
+}
+
+
+function FabricCarToyota (model, year, color, maxSpeed, fuelCapacity, fuelConsumption){
+	Car.apply(this, arguments);
+}
+
+FabricCarToyota.prototype = Car.prototype;
```

Comment:
It is not a good way just to link `Car.prototype` to `FabricCarToyota.prototype`, because it will be one object. It is better to use `Object.create()` function. You can read about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create). Also, then you need to rewrite the `constructor` link to initial